orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/persistence/jedis/JedisExecutionRepository.java

```diff
@@ -0,0 +1,649 @@
+          return orchestration;
+        }
+        correlationRedis.del(key);
+      }
+      throw new ExecutionNotFoundException(
+        format("No Orchestration found for correlation ID %s", correlationId)
+      );
+    });
+  }
+
+  private void storeExecutionInternal(Transaction tx, Execution execution) {
+    tx.sadd(alljobsKey(execution.getType()), execution.getId());
+    tx.sadd(appKey(execution.getType(), execution.getApplication()), execution.getId());
+
+    String key = format("%s:%s", execution.getType(), execution.getId());
+    String indexKey = format("%s:stageIndex", key);
+
+    HashMap<String, String> map = new HashMap<>();
+    try {
+      map.put("application", execution.getApplication());
+      map.put("canceled", String.valueOf(execution.isCanceled()));
+      map.put("limitConcurrent", String.valueOf(execution.isLimitConcurrent()));
+      map.put("buildTime", String.valueOf(execution.getBuildTime() != null ? execution.getBuildTime() : 0L));
+      map.put("startTime", execution.getStartTime() != null ? execution.getStartTime().toString() : null);
+      map.put("endTime", execution.getEndTime() != null ? execution.getEndTime().toString() : null);
+      map.put("status", execution.getStatus().name());
+      map.put("authentication", mapper.writeValueAsString(execution.getAuthentication()));
+      map.put("paused", mapper.writeValueAsString(execution.getPaused()));
+      map.put("keepWaitingPipelines", String.valueOf(execution.isKeepWaitingPipelines()));
+      if (execution.getExecutionEngine().name() != null) {
+        map.put("executionEngine", String.valueOf(execution.getExecutionEngine().name()));
+      } else {
+        map.put("executionEngine", DEFAULT_EXECUTION_ENGINE.name());
+      }
+      map.put("origin", execution.getOrigin());
+      map.put("trigger", mapper.writeValueAsString(execution.getTrigger()));
+      map.put("stageIndex", execution.getStages().stream().map(Stage::getId).collect(Collectors.joining(",")));
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Failed serializing execution", e);
+    }
+    // TODO: remove this and only use the list
+    if (!execution.getStages().isEmpty()) {
+      tx.del(indexKey);
+      tx.rpush(indexKey, execution.getStages().stream()
+        .map(Stage::getId)
+        .collect(Collectors.toList())
+        .toArray(new String[0])
+      );
+    }
+    execution.getStages().forEach(s -> map.putAll(serializeStage(s)));
+
+    if (execution.getType() == PIPELINE) {
+      try {
+        map.put("name", execution.getName());
+        map.put("pipelineConfigId", execution.getPipelineConfigId());
+        map.put("notifications", mapper.writeValueAsString(execution.getNotifications()));
+        map.put("initialConfig", mapper.writeValueAsString(execution.getInitialConfig()));
+      } catch (JsonProcessingException e) {
+        throw new RuntimeException("Failed serializing execution", e);
```

Comment:
Can we use a better exception type? I know this is probably the fault of whoever wrote the old Groovy code…

Reply:
Will go over all of the exceptions thrown in this PR and implement new types when there isn't already a good specific fit.